### PR TITLE
[JBIDE-19179] Add defered loading for resources and add connection refresh

### PR DESCRIPTION
**Pull request by jcantrill**
@adietish Add defered loading and refresh

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.common.ui/build.properties [hunk 1]
```diff
@@ -2,4 +2,7 @@ source.. = src/
 output.. = bin/
 bin.includes = META-INF/,\
                .,\
-               plugin.xml
+               plugin.xml,\
+               plugin.properties,\
+               icons/,\
+               pom.xml
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.common.ui/build.properties]: why would you want to add the pom to a binary artifact? wont hurt but not required

File: plugins/org.jboss.tools.openshift.common.ui/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -18,13 +19,15 @@ Require-Bundle: org.jboss.tools.openshift.common.core;bundle-version="1.0.0",
...
- org.eclipse.ui.navigator;bundle-version="3.5.500"
+ org.eclipse.ui.navigator;bundle-version="3.5.500",
+ org.jboss.tools.openshift.client;bundle-version="3.0.0"
 Bundle-RequiredExecutionEnvironment: JavaSE-1.6
 Bundle-ActivationPolicy: lazy
 Bundle-Vendor: %Bundle-Vendor
 Export-Package: org.jboss.tools.openshift.internal.common.ui;x-friends:="org.jboss.tools.openshift.ui,org.jboss.tools.openshift.express.ui",
  org.jboss.tools.openshift.internal.common.ui.connection;x-friends:="org.jboss.tools.openshift.ui,org.jboss.tools.openshift.express.ui",
  org.jboss.tools.openshift.internal.common.ui.databinding;x-friends:="org.jboss.tools.openshift.ui,org.jboss.tools.openshift.express.ui",
+ org.jboss.tools.openshift.internal.common.ui.explorer,
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.common.ui/META-INF/MANIFEST.MF]: if you export an internal package you'd most likely want to add the x-friends

File: plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/command/RefreshResourceHandler.java [hunk 3]
```diff
@@ -18,20 +20,28 @@
...
+//import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+
+
+
+
+
 
 import com.openshift.client.OpenShiftException;
-import com.openshift.client.Refreshable;
+import com.openshift.client.IRefreshable;
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/command/RefreshResourceHandler.java]: I am missing IRefreshable in the client library. What PR did I miss?

Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/command/RefreshResourceHandler.java]: @adietish fixing now.  I renamed it for consistency, but i dont see it pushed to the server.  Hoping I didnt miss something else i changed

File: plugins/org.jboss.tools.openshift.common.ui/build.properties (hunk 1 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.common.ui/build.properties]: @adietish fixing.  side-effect of some other changes...must have been trigger happy

**Comment by jcantrill** [comment]
@adietish Updated with your changes and added the right changes to OSJC

**Pull request merged**